### PR TITLE
Publish ARM images to preprod env

### PR DESCRIPTION
**Pull request by peterdeme**
## Description of the change

Publish ARM images - for now, only on preprod. If it works, I'll copy it over to prod as well.

This PR must be merged first https://github.com/spacelift-io/runner-terraform/pull/28

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue);
- [x] New feature (non-breaking change that adds functionality);
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected);
- [ ] Documentation (change not affecting any of the images);

### Development

- [ ] The affected image builds in your local environment;

### Code review

- [ ] This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached;
- [ ] This pull request is no longer marked as "draft";
- [ ] Reviewers have been assigned;
- [ ] Changes have been reviewed by at least one other engineer;

### CODE REVIEW

File: dotnet/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,15 @@
...
 
 USER root
 RUN apt-get install -y icu-devtools
 USER spacelift
-RUN wget https://dot.net/v1/dotnet-install.sh
+
+RUN wget -q https://dot.net/v1/dotnet-install.sh
 RUN chmod 777 dotnet-install.sh
-RUN ./dotnet-install.sh -c Current
+RUN ./dotnet-install.sh -c STS
```
Comment (peterdeme) [inline dotnet/Dockerfile:11]: `Current` is deprecated, `STS` is the new (Standard Term Support)

**Review by adamconnelly — APPROVED** [review]
The changes look fine to me, but I think it's worth someone with Pulumi knowledge also taking a look (e.g. Kuba, James or Michiel).

**Comment by peterdeme** [comment]
@adamconnelly thanks, I added them

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+env:
+  # The Pulumi version doesn't matter much, it's just used for sanity testing
+  # but worth to update every once in a while
+  PULUMI_VERSION: "v3.46.0"
+  REPOSITORY_BASE_PATH: spaceliftbuild
+
+jobs:
+  build:
+    if: ${{ github.ref_type != 'tag' }}
+    name: 🚧 ${{ matrix.lang }} on ${{ matrix.arch}}
```
Comment (Yantrio) [inline .github/workflows/build.yml:20]: tiny tiny tiny tiny nitpick: this spacing is off around these curly braces.
```suggestion
    name: 🚧 ${{ matrix.lang }} on ${{ matrix.arch }} 
```

**Review by Yantrio — APPROVED** [review]

**Comment by peterdeme** [comment]
@Yantrio I just squashed the commits so that github doesn't complain about commit verification

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v2
+        if: matrix.arch == 'arm64'
+        with:
+          platforms: linux/arm64
+
+      # Building doesn't work with buildx here because of
+      # a stupid behavior:
+      # https://github.com/moby/moby/issues/42893#issuecomment-1241274246
```
Comment (adamconnelly) [inline .github/workflows/build.yml:45]: Random question - is it worth us using a multistage build instead and continuing to use buildx? Or will that not work in this situation?

**Review by adamconnelly — APPROVED** [review]
LGTM. I left a question, but you're in a much better position than me to answer it so if it doesn't make sense just ignore it.

File: .github/workflows/build.yml (hunk 3 above)
Comment (peterdeme) [reply .github/workflows/build.yml:45]: yeah I think this could be refactored, but in the future sometime. I just want to get this over with for now.

Comment (adamconnelly) [reply .github/workflows/build.yml:45]: No worries.

**Pull request merged**